test/unit.test.js

```diff
@@ -197,14 +235,34 @@ for (const { testName, isRoot } of unitTests) {
         fs.writeFileSync(join(unitPath, 'actual.js'), JSON.stringify(sortedFileList, null, 2));
         throw e;
       }
+
+      if (cached) {
+        // Everything should be cached in the second run, except for `processed-dependency` which adds 1 new input file
+        expect(stat).toHaveBeenCalledTimes(0);
+        expect(readlink).toHaveBeenCalledTimes(testName === 'processed-dependency' ? 1 : 0);
+        expect(readFile).toHaveBeenCalledTimes(testName === 'processed-dependency' ? 1 : 0);
+        expect(analyze).toHaveBeenCalledTimes(testName === 'processed-dependency' ? 1 : 0);
+      } else {
+        // Ensure all cached calls are only called once per file. The expected count is the count of calls unique per path
+        const uniqueStatCalls = [...new Set(stat.mock.calls.map((call) => call[0]))].length;
+        const uniqueReadlinkCalls = [...new Set(readlink.mock.calls.map((call) => call[0]))].length;
+        const uniqueReadFileCalls = [...new Set(readFile.mock.calls.map((call) => call[0]))].length;
+        const uniqueAnalyzeFileCalls = [...new Set(analyze.mock.calls.map((call) => call[0]))].length;
```

Comment:
`Set` has a `.size` property you can use instead of `.length`

Reply:
done